src/Utilities/ConstantExpressions.hpp

```diff
@@ -89,45 +90,83 @@ constexpr uint64_t factorial(const uint64_t n) noexcept {
 
 /// \ingroup ConstantExpressionsGroup
 namespace ConstantExpressions_detail {
-template <typename T, int N, typename = std::nullptr_t>
-struct pow;
 
-/// \cond HIDDEN_SYMBOLS
-template <typename T, int N>
-struct pow<T, N, Requires<(N > 0)>> {
-  SPECTRE_ALWAYS_INLINE static constexpr decltype(auto) apply(const T& t) {
-    return t * pow<T, N - 1>::apply(t);
-  }
-};
+// Implementation functions for the pow template function below which computes
+// optimized powers where the exponent is a compile-time integer
 
-template <typename T, int N>
-struct pow<T, N, Requires<(N < 0)>> {
-  SPECTRE_ALWAYS_INLINE static constexpr decltype(auto) apply(const T& t) {
-    return static_cast<T>(1) / (t * pow<T, -N - 1>::apply(t));
-  }
-};
+// base case power 0: Returns 1.0. This execution path will only be followed if
+// called by the recursive power assembling functions below. This will need to
+// be overloaded for types for which the simple return is inappropriate.
+template <typename T>
+SPECTRE_ALWAYS_INLINE constexpr decltype(auto) pow_impl(
+    const T& /*t*/, std::integral_constant<int, 0> /*meta*/,
+    cpp17::bool_constant<true> /*meta*/) noexcept {
+  return 1.0;
+}
+// duplicate required for both execution paths, and template can't be used to
+// allow for priority in overload resolution
+template <typename T>
+SPECTRE_ALWAYS_INLINE constexpr decltype(auto) pow_impl(
+    const T& /*t*/, std::integral_constant<int, 0> /*meta*/,
+    cpp17::bool_constant<false> /*meta*/) noexcept {
```

Comment:
The recursion between the various `pow_impl` functions is not working the way you want, resulting in extra multiplications for N not 0, 1, or -2.  This overload should be removed, ...